### PR TITLE
function removeCorrespondentDevice added

### PR DESCRIPTION
**Pull request by vakar8888**
executes a delete statement on table correspondent_devices; are there any other tables involved? 
Table device_messages was always empty during testing.
The new function is used in byteball wallet to remove paired devices.

### CODE REVIEW

File: wallet.js [hunk 1]
```diff
@@ -1086,6 +1093,18 @@ function sendMultiPayment(opts, handleResult)
 	});
 }
 
+function readDeviceAddressesUsedInSigningPaths(onDone){
+	db.query(
+		"SELECT distinct device_address FROM shared_address_signing_paths UNION SELECT distinct device_address FROM wallet_signing_paths", 
+		function(rows){
+			
+			var arrDeviceAddress = rows.map(e => e.device_address);
```
Comment (tonyofbyteball) [inline wallet.js]: not sure ES6 will work under browserify/cordova

File: wallet.js [hunk 2]
```diff
@@ -1086,6 +1093,18 @@ function sendMultiPayment(opts, handleResult)
 	});
 }
 
+function readDeviceAddressesUsedInSigningPaths(onDone){
+	db.query(
+		"SELECT distinct device_address FROM shared_address_signing_paths UNION SELECT distinct device_address FROM wallet_signing_paths",
```
Comment (tonyofbyteball) [inline wallet.js]: please use uppercase for sql keywords (DISTINCT)
yes add pending_shared_... even though they are not used now

File: wallet.js [hunk 3]
```diff
@@ -147,7 +147,14 @@ function handleMessageFromHub(ws, json, device_pubkey, bIndirectCorrespondent, c
 			eventBus.emit("text", from_address, body);
 			callbacks.ifOk();
 			break;
-		
+
+		case "remove_paired_device":
+			device.removeCorrespondentDevice(from_address, function(){
+				eventBus.emit("remove_paired_device", from_address, body);
```
Comment (tonyofbyteball) [inline wallet.js]: should be called remove*d*_paired_device since it is an event

**Comment by vakar8888** [comment]
I've closed the duplicated pull request; all changed files are in this pull request

What about the 'trading bot' problem you mentioned? To solve this we need e.g. another flag to distinguish such devices from simple chat devices, or don't we? If I should look for a solution, please tell me which kinds of devices can be paired and how unlinking should be handled (e.g. handshake, subscribing/unsubscribing at bots,...)

**Comment by tonyofbyteball** [comment]
Yes I think we should have a flag in conf (e.g. bIgnoreUnpairRequests) that says that we should ignore removed_paired_device commands received from peers.

**Comment by vakar8888** [comment]
This solution still leaves the problem, that a peer can remove a paired device (e.g. the bot) and after that will not receive any messages from the bot. The bot still sends messages because it ignored the removed_paired_device message.

Another solution:
- device A accepts invitation from device B (bot)
- device B sends pairing message to device A; in the message body it sends a flag, that device A must unsubscribe before removing device B from devices list
- then, device B is not removable, but a 'unsubscribe' button will appear
- when device A unsubscribes, a new message will be sent ('unsubscribing') to device B
- then device B decides whether unsubscribing is allowed or not and sends an answer
- device A receives the answer message and can handle it accordingly

Problem: what happens if device B never answers?
Solution:
- wallet GUI allows hiding and unhiding of devices, so a user can decide to remove a device from the chat list, but messages will still be received. If necessary or wanted the device can be shown in chat list again.
- messages from hidden devices will be shown on demand

**Comment by tonyofbyteball** [comment]
Sorry for late response, github notifications somehow go unnoticed.

There is one problem with the proposed solution that it can be abused by device B to never allow unsubscribing itself.

Actually, I don't see any issues in bot's messages not being delivered.  It is a bot, there are no expectations, no feelings, if its messages are not being replied to.  It is not the case for humans, but we won't allow to set bIgnoreUnpairRequests flag in the GUI wallet.

**Comment by vakar8888** [comment]
Sorry for late response. Other work was important.

OK. I'll add the bIgnoreUnpairRequests flag; if set to true, the removed_paired_device will be ignored.
How can setting the flag be avoided in GUI wallet conf? Overwriting at startup?
Should there be a default?

**Comment by tonyofbyteball** [comment]
OK.
It should default to false, will stay false in GUI wallets, and we don't expose this setting in the UI.  If a user edits it manually, he should know what he is doing.

**Comment by vakar8888** [comment]
flag bIgnoreUnpairRequests added

**Pull request merged**